front/src/layouts/Todo.vue

```diff
@@ -0,0 +1,90 @@
+    <todo-new v-if="new_todo"></todo-new>
+    <todo-edit :todo='to_edit_todo' v-if="edit_todo"></todo-edit>
+  </div>
+</q-layout>
+</template>
+
+<script>
+import Todo from '../components/Todo.vue';
+import TodoHeader from '../components/todo/todo-header.vue';
+import TodoCreateButton from '../components/todo/todo-create-button.vue'
+import TodoList from '../components/todo/todo-list.vue'
+import TodoNew from '../components/todo/todo-new.vue'
+import TodoEdit from '../components/todo/todo-edit.vue'
+
+export default {
+  components: { Todo, TodoHeader, TodoCreateButton, TodoList, TodoNew, TodoEdit },
+  name: "TreeTable",
+  data(){
+    return {
+      edit_todo: false,
+      new_todo: false,
+      to_edit_todo: null,
+      edit_index: null,
+      todos: [
+          {
+              completed: false,
+              title: 'Create a Todo List',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+          {
+              completed: false,
+              title: 'Visit TV Anchor',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+          {
+              completed: false,
+              title: 'Share campaigns to social media',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+          {
+              completed: false,
+              title: 'Create a Todo List',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+          {
+              completed: false,
+              title: 'Create a Todo List',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+      ]
+    }
+  },
+  created(){
```

Comment:
Setting up root listeners is less than ideal.  Instead, on each component, you can create @eventName listeners.

Also, adding spaces to event names is a bad idea.

Also, not using scoped events will certainly cause conflicts in apps.  For example, a root event for a todo page called "create new" will certainly conflict with "create new" in any other page where you would create something new, as there is no indication that this belongs to the todo page.  Something like "todo.create" would be much more sustainable. However, emmiting on the root bus should always be avoided unless you need to, and in this case you don't even need to.